src/System.Text.Json/src/System/Text/Json/JsonUtf8Reader.cs

```diff
@@ -0,0 +1,1274 @@
+            consumed = i;
+            return true;
+        }
+
+        private ConsumeNumberResult ConsumeNegativeSign(ref ReadOnlySpan<byte> data, ref int i)
+        {
+            byte nextByte = data[i];
+
+            if (nextByte == '-')
+            {
+                i++;
+                if (i >= data.Length)
+                {
+                    if (IsLastSpan)
+                    {
+                        _lineBytePosition += i;
+                        ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedDigitNotFoundEndOfData, nextByte);
+                    }
+                    else
+                        return ConsumeNumberResult.NeedMoreData;
+                }
+
+                nextByte = data[i];
+                if ((uint)(nextByte - '0') > '9' - '0')
+                {
+                    _lineBytePosition += i;
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedDigitNotFound, nextByte);
+                }
+            }
+            return ConsumeNumberResult.OperationIncomplete;
+        }
+
+        private ConsumeNumberResult ConsumeZero(ref ReadOnlySpan<byte> data, ref int i)
+        {
+            i++;
+            byte nextByte = default;
+            if (i < data.Length)
+            {
+                nextByte = data[i];
+                if (JsonConstants.Delimiters.IndexOf(nextByte) >= 0)
+                    return ConsumeNumberResult.Success;
+            }
+            else
+            {
+                if (IsLastSpan)
+                    return ConsumeNumberResult.Success;
+                else
+                    return ConsumeNumberResult.NeedMoreData;
+            }
+            nextByte = data[i];
+            if (nextByte != '.' && nextByte != 'E' && nextByte != 'e')
+            {
+                _lineBytePosition += i;
+                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedNextDigitComponentNotFound, nextByte);
+            }
+
+            return ConsumeNumberResult.OperationIncomplete;
+        }
+
+        private ConsumeNumberResult ConsumeIntegerDigits(ref ReadOnlySpan<byte> data, ref int i)
+        {
+            byte nextByte = default;
+            for (; i < data.Length; i++)
+            {
+                nextByte = data[i];
+                if ((uint)(nextByte - '0') > '9' - '0')
+                    break;
+            }
+            if (i >= data.Length)
+            {
+                if (IsLastSpan)
+                    return ConsumeNumberResult.Success;
+                else
+                    return ConsumeNumberResult.NeedMoreData;
+            }
+            if (JsonConstants.Delimiters.IndexOf(nextByte) >= 0)
+                return ConsumeNumberResult.Success;
+
+            return ConsumeNumberResult.OperationIncomplete;
+        }
+
+        private ConsumeNumberResult ConsumeDecimalDigits(ref ReadOnlySpan<byte> data, ref int i, byte nextByte)
```

Comment:
`nextByte` is never read (and is always '.'), why is it passed in?

Reply:
Oh, I see, it's read for the exception.  But it's still always '.' :)

Reply:
> But it's still always '.' :)

Not always, but for the first throw, yes.

I can change it and use a local for the nextByte that is different than '.'. Though, not sure if there is much of an advantage.